src/educational-technology.rst

```diff
@@ -1,18 +1,227 @@
+
+  .. bibliography:: educational-technology.bib
+    :filter: key == 'maloney2010scratch'
+
+  This paper gives an overview of the design of the Scratch programming environment. Scratch is a graphical, block-based programming environment
+  designed for kids who have no programming experience. The system strives to make execution visible and allow for tinkering (learning by trying). The
+  code is live such that any code fragment can be executed simply by clicking on it (no compile-link-run cycle to go through).
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology2.bib
+    :filter: key == 'maloney2008programming'
+
+  Scratch was introduced as an optional activity at an after-school technology center for traditionally disadvantaged youth in 
+  an impoversished area. Little supervision or formal teaching was provided for the system. Over the course of two years,
+  the youth used Scratch more than any other media-creation tool available at the center. They felt that Scratch
+  was most similar to school activities that particularly supported creativity and personal expression, as opposed 
+  to subject areas that are typically associated with programming.
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology2.bib
+    :filter: key == 'wilson2010evaluating'
+
+  Scratch was used for teaching lessons in an IT class for 8-9 year old students at a school. The study
+  did not find evidence of whether or not using this programming environment helped the students develop cognitive abilities. 
+  However, based on feedback from the students, they generally enjoyed and were enthusiastic about their lessons 
+  that used Scratch. 
+
 Feedback Generation
 ===================
 
+With the growing number of people interested in learning programming skills, institutions have difficulty maintaining the number of instructors
+needed to give students valuable one-on-one feedback. Thus, automatic feedback generation is innvaluable in the current learning climate.
+
+There are various ways for students to get feedback. One way that automatic feedback is being generated is in systems that perform as "autograders" where
+students can upload submissions and get information back on how their solutions performed against test suites. This however still leaves a burden
+on students to search out where their solution went wrong, which may be difficult for novices. Thus, feedback that can guide students through
+debugging code with errors can be far more valuable to the learning process than what is returned by standard autograders.
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology.bib
+    :filter: key == 'suzuki2017exploring'
+
+  Five types of common hints that teachers give to help students fix their code that can be generated using program synthesis
+  are itentified: transformations (what to change to make the program work), locations (the line(s) that need to be changed to make the program work),
```

Comment:
consider making these numbered lists rather than leaving them inline in the sentence